src/transformers/models/molmo/modular_molmo.py

```diff
@@ -0,0 +1,2289 @@
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        hidden_states = self.embeddings(pixel_values)
+        hidden_states = self.pre_layrnorm(hidden_states)
+
+        encoder_outputs = self.encoder(
+            inputs_embeds=hidden_states,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        last_hidden_state = encoder_outputs[0]
+
+        if not return_dict:
+            return (last_hidden_state) + encoder_outputs[1:]
+
+        return BaseModelOutput(
+            last_hidden_state=last_hidden_state,
+            hidden_states=encoder_outputs.hidden_states,
+            attentions=encoder_outputs.attentions,
+        )
+
+
+class MolmoVisionModel(CLIPVisionModel):
+    pass
+
+
+class MolmoPoolingAttention(nn.Module):
+    """Multi-headed attention from 'Attention Is All You Need' paper"""
+
+    def __init__(self, config):
+        super().__init__()
+        self.config = config
+        self.embed_dim = config.hidden_size
+        self.num_heads = config.num_attention_heads
+        self.head_dim = config.head_dim
+
+        self.scale = self.head_dim**-0.5
+        self.dropout = config.attention_dropout
+
+        self.k_proj = nn.Linear(self.embed_dim, self.num_heads * self.head_dim)
+        self.v_proj = nn.Linear(self.embed_dim, self.num_heads * self.head_dim)
+        self.q_proj = nn.Linear(self.embed_dim, self.num_heads * self.head_dim)
+        self.o_proj = nn.Linear(self.num_heads * self.head_dim, self.embed_dim // 2)
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        key_value_hidden_states: torch.Tensor,
+        output_attentions: Optional[bool] = False,
+    ) -> Tuple[torch.Tensor, Optional[torch.Tensor]]:
+        """Input shape: Batch x Time x Channel"""
+
+        bsz, tgt_len, embed_dim = hidden_states.size()
+        seq_len = key_value_hidden_states.shape[1]
+        query_states = self.q_proj(hidden_states) * self.scale
+        key_states = (
+            self.k_proj(key_value_hidden_states)
+            .view(bsz, seq_len, self.num_heads, self.head_dim)
+            .transpose(1, 2)
+            .contiguous()
+        )
+        value_states = (
+            self.v_proj(key_value_hidden_states)
+            .view(bsz, seq_len, self.num_heads, self.head_dim)
+            .transpose(1, 2)
+            .contiguous()
+        )
+
+        proj_shape = (bsz * self.num_heads, -1, self.head_dim)
+        query_states = (
+            query_states.view(bsz, tgt_len, self.num_heads, self.head_dim)
+            .transpose(1, 2)
+            .contiguous()
+            .view(*proj_shape)
+        )
```

Comment:
let's rather look at how we have this standardized! -> split this in 2 lines to make ti more readable please!